ONSdigital/dp-filter-api
# Feature/validate filter dimensions

### What

Validate filter dimensions and dimension options by calling the dataset API to check whether they exist, if they don't exist then return bad request and message detailing the fields that are wrong. There is future work to return error objects in json instead of a message.

### How to review

Check logic makes sense and that all endpoints (PUT's and POST's) that change a filter blueprint validate request. Any TODO's and FIXME's are to use new endpoints that should be available in future dataset API work.

Make sure unit tests work.

Run acceptance tests by going to `feature/validate-filter-dimensions` on the dp-api-tests branch; then do the following:

1) Run dp-dataset-api
2) Run dp-filter-api (on `feature/validate-filter-dimensions`)
3) cd filterAPI; go test ./...

### Who can review

Team B
